llvm/tools/llvm-objdump/llvm-objdump.cpp

```
@@ -3533,6 +3533,51 @@ commaSeparatedValues(const llvm::opt::InputArgList &InputArgs, int ID) {
   return Values;
 }
 
+static void mcpuHelp() {
+  Triple TheTriple;
+
+  if (!TripleName.empty()) {
+    TheTriple.setTriple(TripleName);
+  } else {
+    assert(!InputFilenames.empty());
+    Expected<OwningBinary<Binary>> OBinary = createBinary(InputFilenames[0]);
+    if (Error E = OBinary.takeError()) {
+      reportError(InputFilenames[0], "triple was not specified and could not "
+                                     "be inferred from the input file: " +
+                                         toString(std::move(E)));
+    }
+
+    Binary *Bin = OBinary->getBinary();
+    if (ObjectFile *Obj = dyn_cast<ObjectFile>(Bin)) {
+      TheTriple = Obj->makeTriple();
+    } else if (Archive *A = dyn_cast<Archive>(Bin)) {
+      Error Err = Error::success();
+      for (auto &C : A->children(Err)) {
+        Expected<std::unique_ptr<Binary>> ChildOrErr = C.getAsBinary();
+        if (!ChildOrErr)
+          // We don't report error because we don't disassemble it here.
```

I'm not sure I follow why this case is different to the case earlier at https://github.com/llvm/llvm-project/blob/7904d355b34b8ae2a9ead825f7fe57f4c6eff803/llvm/tools/llvm-objdump/llvm-objdump.cpp#L3544? Why do we ignore the error in this case instead of reporting that we couldn't infer the triple from the object file?

I also note that this case doesn't have any test coverage.